src/handlers/actions/EmitDomainReputation.ts

```diff
@@ -0,0 +1,43 @@
+import { BigNumber } from 'ethers';
+
+import { Id, getEvents } from '@colony/colony-js';
+import networkClient from '~/networkClient';
+import { ColonyActionType, ContractEvent } from '~/types';
+import { toNumber, writeActionFromEvent, getDomainDatabaseId } from '~/utils';
+
+export default async (event: ContractEvent): Promise<void> => {
+  const { contractAddress: colonyAddress } = event;
+  const { agent: initiatorAddress, user: userAddress, skillId, amount } = event.args;
+
+  const isReputationChangePositive = BigNumber.from(amount).gt(0);
+
+  const actionType = isReputationChangePositive ? ColonyActionType.EmitDomainReputationReward : ColonyActionType.EmitDomainReputationPenalty;
+
+  const colonyClient = await networkClient.getColonyClient(colonyAddress);
+  const domainAddedFilter = colonyClient.filters.DomainAdded(null, null);
+  const domainAddedEvents = await getEvents(colonyClient, domainAddedFilter);
+
+  const colonyDomains = await Promise.all(
+    domainAddedEvents.map(async (domain) => {
+      const domainId = domain.args.domainId.toString();
+      const { skillId } = await colonyClient.getDomain(domainId);
+      return {
+        skillId,
+        domainId,
+      };
+    }),
+  );
```

Comment:
An alternative solution would be to query the database for the domains we store there. I don't know which one is more performant. Just something to look out in the future

Reply:
@jakubcolony do you think we should create an issue for this to investigate? or should the current solution be fine for now.